bin/assets/scripts/loc.js

```diff
@@ -43,3 +45,25 @@ function handleHash() {
     }
   }
 }
+
+if (token) {
+  deleteButton.addEventListener('click', () => {
+    fetch(window.location.pathname, {
+      method: 'DELETE',
+      headers: {
+        Authorization: `Token ${token}`
+      }
+    })
+      .then((res) => {
+        if (!res.ok) {
+          console.error(res);
+          throw new Error(res.statusText);
+        }
+        window.location.href = '/';
+      })
+      .catch((err) => {
+        alert(`Une erreur est survenue (${err.message}), votre snippet n'a donc pas pu être supprimé. Regardez la console pour plus d'informations.`);
+        console.error("Nous vous recommendons d'ouvrir une issue sur https://github.com/readthedocs-fr/bin-server si le problème persiste.", err);
+      });
+  });
+}
```

Comment:
C'est dommage d'avoir *besoin* de javascript pour cette fonctionnalité, est-ce que ce ne serait pas possible d'avoir un formulaire avec méthode `DELETE` plutôt ? Le fait qu'on passe par le header Authorization risque d'être problématique mais j'aimerais qu'on explore cette possibilité (je sais qu'on a des users avec noscript)

Reply:
Il suffit d'ajouter un field supplémentaire contenant la "vraie" méthode et d'en récupérer la valeur au niveau du traitement.
Bon nombre de frameworks web permettent ceci

Reply:
c'est-a-dire ?

Reply:
Dans le formulaire au niveau html

Reply:
Pour référence <https://softwareengineering.stackexchange.com/questions/114156/why-are-there-are-no-put-and-delete-methods-on-html-forms> (TL;DR: le W3C n'a jamais pris la peine de bosser dessus, donc ça supporte que GET et POST, oui c'est de la merde) mais le problème reste entier, c'est une fonctionnalité intéressante que les utilisateurs noscript n'auront pas moyen d'utiliser. Est-ce que ce serait envisageable de documenter qqpart de lisible pour l'utilisateur la commande `cron` équivalente pour supprimer le bin ?

Reply:
@Hokkaydo même en passant par un framework le problème reste entier (voir pire) : on parle des utilisateurs noscript ici, les mêmes qui ne pourront pas charger le-dit framework.

Reply:
Mmh je vois
Tu as une autre idée dans ce cas ?